exercises/23_csv.rb

```diff
@@ -0,0 +1,23 @@
+require "csv" 
+
+employees = Hash.new()
+# File.open("23.csv", "r") do |file|
+#   file.gets
+#   while line = file.gets
+#     name, id, designation = line.split(',').map { |el| el.strip }
+#     employees[:"#{designation}"] = employees[:"#{designation}"] || []
+#     employees[:"#{designation}"] << "#{name} (EmpId: #{id})"
+#   end
+# end
+
+CSV.foreach("23.csv", headers: true) do |row|
+  employees[:"#{row["Designation"]}"] = employees[:"#{row["Designation"]}"] || []
```

Comment:
No need to have this line. You can do the same thing by just replacing line 3 with
`employees = Hash.new { |h,k| h[k] = [] }`
This will initiate the hash with array